cli/pkg/cmd/upgrade/upgrade.go

```diff
@@ -0,0 +1,110 @@
+package upgrade
+
+import (
+	"cli/pkg/cmd/config"
+	"cli/pkg/console"
+	"cli/pkg/helm"
+	"cli/pkg/kube"
+	"cli/pkg/workspace"
+	"fmt"
+
+	"github.com/airyhq/airy/infrastructure/lib/go/k8s/util"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var (
+	approve    bool
+	version    string
+	Version    string
+	UpgradeCmd = &cobra.Command{
+		Use:   "upgrade [workspace directory]",
+		Short: "Upgrades an instance of Airy Core",
+		Long:  `Upgrades an existing Airy Core instance, read from a workspace directory (default .)`,
+		Args:  cobra.MaximumNArgs(1),
+		Run:   upgrade,
+	}
+)
+
+func init() {
+	UpgradeCmd.Flags().StringVar(&version, "version", "", "Specify a version to upgrade to (optional).")
+	UpgradeCmd.Flags().BoolVar(&approve, "approve", false, "Upgrade automatically without asking for an approval (optional).")
+}
+
+func upgrade(cmd *cobra.Command, args []string) {
+	workspacePath := ""
+	if len(args) > 0 {
+		workspacePath = args[0]
+	}
+
+	dir := workspace.Init(workspacePath)
+	conf, err := dir.LoadAiryYaml()
+	if err != nil {
+		console.Exit("Error parsing configuration file: ", err)
+	}
+	kubeCtx := kube.Load()
+	clientset, err := kubeCtx.GetClientSet()
+	namespace := viper.GetString("namespace")
+	if err != nil {
+		console.Exit("could not find an installation of Airy Core. Get started here https://airy.co/docs/core/getting-started/installation/introduction")
+	}
+
+	if version == "" {
+		version = Version
+	}
+	oldVersion := conf.Kubernetes.AppImageTag
```

Comment:
Not sure if we should read the current version from the `airy.yaml` file or from the `airy-config-map` configMap, inside the Kubernetes cluster.

Reply:
The one from the cluster makes more sense to me, since it should reflect the current state. The config file always represents the "should" state.